# Upstream status: in 2.6.18-rc3

commit a6a38a66224c7c578cfed2f584b440c81af0c3ae
Author: Lennert Buytenhek <buytenh@wantstofly.org>
Date:   Sat Jul 29 08:29:26 2006 +0100

    [ARM] 3736/1: xscale: don't mis-report 80219 as an iop32x
    
    Patch from Lennert Buytenhek
    
    The IOP 80219 xscale CPU is a stripped down version of the IOP32x.
    But the fact that the 80219 and IOP32x are very similar doesn't mean
    that they need to share a cpu table entry.  It's also somewhat confusing
    for the end user to see the 80219 reported as an IOP32x, so this patch
    splits the IOP32x cpu table entry to make a separate entry for the
    80219.
    
    Signed-off-by: Lennert Buytenhek <buytenh@wantstofly.org>
    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>


diff -urN a/arch/arm/mm/proc-xscale.S b/arch/arm/mm/proc-xscale.S
--- a/arch/arm/mm/proc-xscale.S	2006-08-03 11:39:01.692744500 +0000
+++ b/arch/arm/mm/proc-xscale.S	2006-08-03 11:44:23.780873750 +0000
@@ -532,6 +532,11 @@
 	.asciz	"XScale-80200"
 	.size	cpu_80200_name, . - cpu_80200_name
 
+	.type	cpu_80219_name, #object
+cpu_80219_name:
+	.asciz	"XScale-80219"
+	.size	cpu_80219_name, . - cpu_80219_name
+
 	.type	cpu_8032x_name, #object
 cpu_8032x_name:
 	.asciz	"XScale-IOP8032x Family"
@@ -606,10 +611,30 @@
 	.long	xscale_cache_fns
 	.size	__80200_proc_info, . - __80200_proc_info
 
+	.type	__80219_proc_info,#object
+__80219_proc_info:
+	.long	0x69052e20
+	.long	0xffffffe0
+	.long   PMD_TYPE_SECT | \
+		PMD_SECT_BUFFERABLE | \
+		PMD_SECT_CACHEABLE | \
+		PMD_SECT_AP_WRITE | \
+		PMD_SECT_AP_READ
+	b	__xscale_setup
+	.long	cpu_arch_name
+	.long	cpu_elf_name
+	.long	HWCAP_SWP|HWCAP_HALF|HWCAP_THUMB|HWCAP_FAST_MULT|HWCAP_EDSP
+	.long	cpu_80219_name
+	.long	xscale_processor_functions
+	.long	v4wbi_tlb_fns
+	.long	xscale_mc_user_fns
+	.long	xscale_cache_fns
+	.size	__80219_proc_info, . - __80219_proc_info
+
 	.type	__8032x_proc_info,#object
 __8032x_proc_info:
 	.long	0x69052420
-	.long	0xfffff5e0      @ mask should accomodate IOP80219 also
+	.long	0xffffffe0
 	.long   PMD_TYPE_SECT | \
 		PMD_SECT_BUFFERABLE | \
 		PMD_SECT_CACHEABLE | \

